### PR TITLE
Active Buzzer

### PR DESCRIPTION
**Pull request by adelgadoj3**
Resolves active buzzer helper and controller. Passive Buzzer needs more work done. Will be a new issue for next sprint.

### CODE REVIEW

File: components/src/main/java/com/opensourcewithslu/components/controllers/ActiveBuzzerController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.opensourcewithslu.components.controllers;
+
+import com.opensourcewithslu.outputdevices.ActiveBuzzerHelper;
+//import com.pi4j.io.gpio.digital.DigitalOutput;
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/ActiveBuzzerController.java:4]: get rid of unnecessary comments

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package com.opensourcewithslu.outputdevices;
+
+
+//import com.pi4j.io.gpio.digital.DigitalOutput;
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:4]: remove this import

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    /**
+     * Turns the active buzzer on by setting the duty cycle is 50 and frequency to 440hz.
+     */
+    //tag::method[]
+    public void activeBuzzerOn(){
+        //end::method[]
+        log.info("Active buzzer is on.");
+
+
+        this.activeBuzzer.on(100, 3000); //Duty Cycle must be 100 and Frequency should be 440.
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:45]: change the frequency to 440

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    public void constantTone(){
+
+        int sleepDuration = 1000; //1000 milliseconds equals 1 second
+        int totalActiveDur = 0;
+        activeBuzzerOff();
+
+        while (true){
+            log.info("Buzzer on.");
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:84]: the log.info message "Buzzer On" comes from activeBuzzerOn(), change the message to "Buzzer turns on for 10 sec"

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            activeBuzzerOff();
+            try{
+                Thread.sleep(sleepDuration);
+                totalActiveDur += sleepDuration;
+            } catch (InterruptedException e){
+                Thread.currentThread().interrupt();
+            }
+            log.info("Buzzer off.");
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:92]: remove this log message

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        //end::method[]
+        log.info("Active Buzzer is off.");
+
+        this.activeBuzzer.off();
+
+        actBuzzCheck = false;
+
+    }
+
+    public void alertTone(){
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:64]: change the function name to beep. add a log info message "Beeps for 2 sec"

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            } catch (InterruptedException e){
+                Thread.currentThread().interrupt();
+            }
+            if (totalActiveDur == 10000){
+                break;
+            }
+        }
+    }
+
+    public void buzzerTest(){
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:105]: change the function name to morseCodeTone

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+        int [] morseCodePi = {200,600,600,200,200,200}; // Durations for .--. .. (pi in Morse)
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:107]: remove the Pi at the end of variable name

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+        int gapDuration = 200; //Gap between the signals
+        activeBuzzerOff();
+
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:110]: add a log message "Playing morse code"

**Review by ruthvikm — CHANGES_REQUESTED** [review]
Test it again on pi and check the buzzer functions and corresponding log messages

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java (hunk 6 above)
Comment (SwindleA) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:64]: needs javadoc comment

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+        try {
+            Thread.sleep(2000); //Wait and alert once for 2 seconds (2000 milliseconds)
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        activeBuzzerOff();
+
+    }
+
+    public void constantTone(){
```
Comment (SwindleA) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:77]: Needs javadoc comment

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java (hunk 7 above)
Comment (SwindleA) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:105]: add javadoc comment

**Review by SwindleA — CHANGES_REQUESTED** [review]
just a few javadoc comments missing. It builds correctly. I assume someone else will test the component.

**Comment by ruthvikm** [comment]
> just a few javadoc comments missing. It builds correctly. I assume someone else will test the component.

We have tested it on pi and it works

File: components/src/main/java/com/opensourcewithslu/components/controllers/ActiveBuzzerController.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+     */
+    @Get("/beepTone")
+    public void playBeepTone(){
+
+        activeBuzzerHelper.beep();
+
+    }
+
+    /**
+     * Emits a constant tone from the active buzzer for a duration of 20 seconds.
```
Comment (ruthvikm) [inline components/src/main/java/com/opensourcewithslu/components/controllers/ActiveBuzzerController.java:50]: This comment doesn't justify the function below

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 12]
```diff
@@ -0,0 +1,143 @@
+package com.opensourcewithslu.outputdevices;
+
+
+
+import com.pi4j.io.pwm.Pwm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * The BuzzerHelper class contains methods that pertain to the control of the active and passive buzzers.
```
Comment (ruthvikm) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:10]: no passive buzzer here

**Review by ruthvikm — CHANGES_REQUESTED** [review]

**Review by ruthvikm — APPROVED** [review]

File: pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java [hunk 13]
```diff
@@ -0,0 +1,143 @@
+package com.opensourcewithslu.outputdevices;
+
+
+
+import com.pi4j.io.pwm.Pwm;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
```
Comment (SwindleA) [inline pi4micronaut-utils/src/main/java/com/opensourcewithslu/outputdevices/ActiveBuzzerHelper.java:9]: This line needs a second '*' for the javadoc to generate. So line 9 needs to be /**

**Review by SwindleA — CHANGES_REQUESTED** [review]
Looks good. One small typo regarding the constructor javadoc.

**Review by SwindleA — APPROVED** [review]
Looks good. Javadocs build correctly

**Pull request merged**